src/service/alerts/scheduler.rs

```diff
@@ -833,35 +828,6 @@ async fn handle_derived_stream_triggers(
         }
     };
 
-    if trigger.retries >= max_retries {
```

Comment:
This block actually never got executed, as triggers with retries ge to max_retries would never have been pulled from scheduler table.